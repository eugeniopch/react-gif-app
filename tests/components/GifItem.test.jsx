import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem />', () => { 

    const titulo = 'Friends';
    const url = 'http://google.com/';

    test('Debe matchear con el snapshot', () => { 

        const { container } = render(<GifItem title={ titulo } url={ url } />);
        expect(container).toMatchSnapshot();

    })

    test('Debe mostrar la imagen con el URL y el ALT indicado', () => { 

        render(<GifItem title={ titulo } url={ url } />);
        //screen.debug();
        
        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( titulo );

        //Manera más eficiente de hacer lo anterior:

        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( titulo );

    })

    test('Debe mostrar el título en el componente', () => { 

        render(<GifItem title={ titulo } url={ url } />);
        expect( screen.getByText( titulo ) ).toBeTruthy();

     })


 })