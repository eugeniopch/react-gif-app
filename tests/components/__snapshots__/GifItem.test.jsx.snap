// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe matchear con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Friends"
      src="http://google.com/"
    />
    <p>
      Friends
    </p>
  </div>
</div>
`;
